.github/scripts/incremental-tag.sh

```diff
@@ -0,0 +1,20 @@
+#! /bin/bash
+set -euo pipefail
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+image=$1
+tag=${2:-$(cd $SCRIPT_DIR/../../ && mvn help:evaluate -Dexpression=project.version -q -DforceStdout)}
+original_tag=$tag
+c=1
+
+while :
+do
+  tag="${original_tag}-${c}"
+  if DOCKER_CLI_EXPERIMENTAL=enabled docker manifest inspect $image:$tag >/dev/null 2>/dev/null; then
+      c=$((c+1))
+  else
+      echo "-${c}"
+      break
+  fi
+done
```

Comment:
This loop doesn't seem necessary.

With anonymous access to the quay API, we can list all the current tags, and determine what the next value for `c` should be from that:

```bash
curl -sSfL 'https://quay.io/api/v1/repository/keycloak/keycloak/tag/?onlyActiveTags=true' | jq -r '.tags | .[] | .name' | head
nightly
latest
18.0
18.0.0
legacy
18.0.0-legacy
17.0.1-legacy
17.0
17.0.1
17.0.0-legacy
```

Reply:
Thanks for the update @ASzc !
This is again a similar tradeoff as the other PR 🙂 , I do believe that plain docker commands are preferable for this task since they work with any Docker registry.
Also, the overall complexity of your proposed solution (although shows great `bash` skills 🙂 ) seems to be a little higher than my proposal.

It would be nice to have a third opinion to break the tie, cc. @abstractj @stianst